stubs/Modifiers/StubsModifier.swift

```diff
@@ -115,80 +115,103 @@ class StubsModifier {
+
+                case .remove(let keys):
+                    for key in keys {
+                        dict[key] = nil
+                    }
+
+                case .transform(let script):
+                    if let object = try? executeScript(script, json: dict) as? [String: Any] {
+                        dict = object
+                    }
+                    else {
+                        printError("Failed to transform dictionary for rule: \(rule.printablePath)")
+                    }
+                }
             }
-        } else {
-            if var dict = object as? [String: Any] {
+
+            return dict as! T
+        } else if let array = object as? [[String: Any]] {
+            let newArray = array.compactMap { (dict) -> [String: Any] in
+                var copy = dict
+
                 for modification in rule.modification {
                     switch modification {
                     case .add(let addDict):
-                        dict.merge(addDict) { (value1, value2) -> Any in
+                        copy.merge(addDict) { (value1, value2) -> Any in
                             return value2
                         }
 
                     case .remove(let keys):
                         for key in keys {
-                            dict[key] = nil
+                            copy[key] = nil
                         }
-                    }
-                }
 
-                return dict as! T
-            } else if let array = object as? [[String : Any]] {
-                let newArray = array.compactMap { (dict) -> [String : Any] in
-                    var copy = dict
-
-                    for modification in rule.modification {
-                        switch modification {
-                        case .add(let addDict):
-                            copy.merge(addDict) { (value1, value2) -> Any in
-                                return value2
-                            }
-
-                        case .remove(let keys):
-                            for key in keys {
-                                copy[key] = nil
-                            }
+                    case .transform(let script):
+                        if let object = try? executeScript(script, json: dict) as? [String: Any] {
+                            return object
+                        }
+                        else {
+                            printError("Failed to transform dictionary for rule: \(rule.printablePath)")
                         }
                     }
-
-                    return copy
                 }
 
-                return newArray as! T
+                return copy
             }
+
+            return newArray as! T
         }
 
-        // This shouldn't happen
         throw ModificationError.incorrectExecutionPath
     }
 
-    private func saveModified(_ modifiedDict: [String: Any], at url: URL) {
+    private func executeScript(_ script: String, json: [String: Any]) throws -> Any {
```

Comment:
Here's the "magic"... if you want to call it that. It takes the JSON object, encodes in base 64 (because I'm lazy and didn't want to be annoyed by spaces and special characters), passes it to the script, and then reads the output from the script with the transformed object.

Reply:
Yeah, why not?